Basic work for right sidebar

PVince81 opened the pull request:
Adds right sidebar with registrable panels (still WIP)

For https://github.com/owncloud/core/issues/17655

~~Note: file actions are currently disabled for demo purposes. When clicking on a file it will show the sidebar but not perform any action.~~
- [x] TODO: rewire click events to have the default file action work on the left side of the row but not the right side
- [x] TODO: implement tabs logic
- [x] TODO: implement default tab
- [x] TODO: styling
- [x] TODO: sidebar must close when unselecting file or selecting multiple files ?

Commit:
Basic work for right sidebar

Adds right sidebar with registrable panels (still WIP)

scrutinizer-notifier commented:
A [new inspection](https://scrutinizer-ci.com/g/owncloud/core/inspections/ae973a9f-a3b7-4eef-a794-f7f981e3ac7f) was created.

PVince81 commented:
Basically this brings the following:
- `DetailsView` which is the right sidebar / panel. On this class it's possible to register subviews
- `DetailFileInfoView` is the base class for any file info block to be displayed in the top panel. Needs to be registered with `detailView.addDetail(...)`
- `MainFileInfoDetailView` implements `DetailFileInfoView` to provide the basic info like thumbnail, file name
- `DetailTabView` is the base class for info tabs. Needs to be registered with `detailView.addTab(...)` (the logic isn't implemented yet)
- styles can be tweaked in `detailsView.css`

I know... a lot of classes and not so fancy names...

PVince81 commented:
@DeepDiver1975 @jancborchardt @blizzz @oparoz @rullzer

PVince81 commented:
@jancborchardt I think it's still too early to tweak the styles. I'll let you know when it's ready :smile:

oparoz commented on apps/files/js/mainfileinfodetailview.js:
```diff
@@ -0,0 +1,75 @@
+				this.$el.append(this._template(this._fileInfo));
+				var $iconDiv = this.$el.find('.thumbnail');
+				// FIXME: use proper way, this is only for demo purposes
+				FileList.lazyLoadPreview({
```
This won't work as there is a hardcoded value of 36x36 as a fallback. We keep running into this issue.
Besides, I don't think we need lazy loading here. Just point to the preview using `generatePreviewUrl`and the browser will show it when ready.

oparoz replied:
The mockup indicates 50x50, but that's too small for the prevew of a selected file. Maybe we should load a 200x200 preview and let the browser resize it based on the available width, if the sidebar is responsive.

PVince81 replied:
I'll add it as todo. The FIXME already says that what I did here is wrong :wink:

oparoz replied:
Ah, yeah, I only looked at the todos :D

PVince81 commented:
- [ ] TODO: load previews with the correct API, maybe with 200x200 format and reduce it to 50x50 ? ( @oparoz feel free to comment/rectify)

Commit:
Improved right sidebar

Added owner info.
Added animation, but causes scrollbal.
Default file action now when clicking on name directly.
Fixed icon.
Added empty share tab.

scrutinizer-notifier commented:
A [new inspection](https://scrutinizer-ci.com/g/owncloud/core/inspections/4bfff24b-c86a-43c9-8c38-e5def50fe2b4) was created.

PVince81 commented:
I realize that the file info details (the top view) cannot be extended block-wise.
Maybe the easiest is to simply pass the jQuery element to plugins so plugins can embed themselves wherever. Here you can see that "Owner: " is in its own block. That one is provided by the "files_sharing" app.

I'm not too happy yet about the way how to provide tab view and detail view, it seems quite cumbersome to implement/extend a big class just for a few lines.

Also due to the problematic loading order of JS files I had to do this: https://github.com/owncloud/core/pull/17656/files#diff-593f07f9e5d32f4d3fdebbfb2fa5b71aR144

Commit:
More fixes to the right sidebar

Sidebar now works in all file list views.
Moved owner to share tab.

PVince81 commented:
- [ ] make it possible to hide tabs based on content type ? (ex: folders)

PVince81 commented:
- [ ] hightlight current row
- [ ] full height scrollbar on the sidebar (let the contents flow, no inner scrollbars)

Commit:
Styling fixes + hightlight in files sidebar

Sidebar scrolls with contents
Highlight currently show file

PVince81 commented:
- [ ] BUG: hiding animation leaves horizontal scrollbar
- [ ] TODO: make favorite action work

PVince81 commented:
- [ ] TODO: unit tests for the panels logic

scrutinizer-notifier commented:
A [new inspection](https://scrutinizer-ci.com/g/owncloud/core/inspections/efcc664a-089f-495d-9b03-1852124c8751) was created.

PVince81 commented:
I'm not sure if we want to do the share dialog task for now.
Maybe we can first get this in a mergable state and the sidebar will initially have no tabs shown at all.
So we need to get the interaction with the file list right.

When testing, please make sure to test the sidebar in the other views: "Shared with you", "Deleted files", etc. :smile:

blizzz commented:
> Basically this brings the following:
- `DetailsView` which is the right sidebar / panel. On this class it's possible to register subviews
- `DetailFileInfoView` is the base class for any file info block to be displayed in the top panel. Needs to be registered with `detailView.addDetail(...)`
- `MainFileInfoDetailView` implements `DetailFileInfoView` to provide the basic info like thumbnail, file name
- `DetailTabView` is the base class for info tabs. Needs to be registered with `detailView.addTab(...)` (the logic isn't implemented yet)
- styles can be tweaked in `detailsView.css`

> I know... a lot of classes and not so fancy names...

Nothing against many classes, if they make the whole feature better readable, understandable and maintainable. Of course this can be done, but I do not see it here.

The naming however should make you roughly understand what this is it about. DetailsView is pretty general, more obvious would be to have Sidebar or Panel in it's name indeed, but would cost you flexibility to change it in the future to a e.g. bumping circle. As it shows details, I am good with the naming.

Most importantly, the structure makes sense to me. All question that occurred to me were well answered in this approach.

PVince81 commented:
@blizzz thanks for the feedback :smile: 
I'd say let's keep this approach then and we can adjust later if needed.

PVince81 commented:
- [ ] fix scrolling: the top of the sidebar must always stay visible when scrolling down the file list (ex: when getting details of a file on page 12). Add a second scrollbar inside the sidebar to scroll its whole content.

PVince81 commented:
![right-sidebar](https://cloud.githubusercontent.com/assets/277525/8704922/7a921ece-2b26-11e5-8a98-99511a4168ad.png)

PVince81 commented:
- [ ] TEST: public page

Commit:
Remove horizontal scroll caused by right sidebar

Commit:
Right sidebar now has its own scrollbar

Commit:
Disable right sidebar in public file list page

Commit:
Improve layout, fix mime icon

scrutinizer-notifier commented:
A [new inspection](https://scrutinizer-ci.com/g/owncloud/core/inspections/0b394f15-3957-4485-b887-2632aef99ab4) was created.

PVince81 commented:
I changed the scrolling behavior: now the sidebar has its own scrollbar.
Unfortunately it doesn't look very good when there are two scrollbars, but there might not be any better way around, unless we add an inner scrollbar inside the tab content instead.

@jancborchardt I think you can start tweaking the styles now and test the interaction :smile: 
I left the debug tabs so you can test the scrolling.

![right-sidebar2](https://cloud.githubusercontent.com/assets/277525/8721295/bf5335dc-2bb9-11e5-9c9a-f30e58ab3eca.png)

PVince81 commented:
- [ ] TEST in internet explorer

PVince81 commented:
- [ ] BUG: fix display name in trash bin (use fileInfo.displayName)

jancborchardt commented:
@PVince81 please _do not_ use jQuery UI here, as we want to ideally get rid of it. Simply use buttons, and I will style them properly similar to the database chooser on installation.

PVince81 commented:
The reason I used tabs is just because I wanted to save time and saw a working example on the LDAP settings page.

- [ ] TODO: implement own tabs plugin/logic

jancborchardt commented:
Ok. Just noting because we definitely want to do it like the db switcher on the install page.

Commit:
Added unit tests for right sidebar

scrutinizer-notifier commented:
A [new inspection](https://scrutinizer-ci.com/g/owncloud/core/inspections/c3a58cef-e43e-4710-a4ee-af3e96b6eaa0) was created.

jancborchardt commented:
@PVince81 @owncloud/designers after chatting with @karlitschek we think it’s really better to have the right sidebar not only overlay the content but really move it away. Otherwise it’s only modal, actions and info below it is hidden.

We also need to pay attention to other apps which use similar things. Calendar+ by @libasys uses a right sidebar for Tasks integration. Same for Tasks by @raimund-schluessler & @Henni – there the right sidebar is for details of a task.
We should define the style and interaction for this sidebar in core so that apps can adapt it.

raimund-schluessler commented:
Interesting work.

I would also vote for moving the content, so you can still access / see actions and info currently below the sidebar. This also solves the scroll bar problem since the scroll bar for the content would then be on the left of the sidebar (see tasks app as example).

oparoz commented:
Bear in mind that some apps need to redraw the entire view when you change the viewport's width. 
If it penalises just Gallery, then maybe that app can work around the problem, but there may be others.

Commit:
More work on right sidebar unit tests

jancborchardt commented:
@oparoz good point, Gallery isn’t fully responsive in this way. Probably also nothing really to do about it than trying to make the redraw faster? Or keep showing the old content while redrawing in the background and replacing it then?

oparoz commented:
Well, Gallery is fully responsive. The problem I see is more a UX one with a redrawn photowall every time a user clicks on the share button. 
This is not really something that one does often, even if each picture can be shared, so maybe a non-issue? I'm just mentioning it in case this could happen to other apps as well.

MorrisJobke commented:
```
16:54:12 PhantomJS 1.9.8 (Linux 0.0.0) OCA.Files.DetailsView tests file info detail view updates registered tabs when fileinfo is updated FAILED
16:54:12 	Expected 4 to equal 2.
16:54:12 	    at /var/jenkins/workspace/pull-request-analyser-ng-simple@12/label/SLAVE/apps/files/tests/js/detailsviewSpec.js:62
16:54:12 .
16:54:12 PhantomJS 1.9.8 (Linux 0.0.0) OCA.Files.DetailsView tests tabs updates registered tabs when fileinfo is updated FAILED
16:54:12 	Expected 4 to equal 2.
16:54:12 	    at /var/jenkins/workspace/pull-request-analyser-ng-simple@12/label/SLAVE/apps/files/tests/js/detailsviewSpec.js:91
16:54:16 PhantomJS 1.9.8 (Linux 0.0.0) OCA.Files.MainFileInfoDetailView tests rendering displays basic info FAILED
16:54:16 	Expected 'July 16, 2015 9:02 PM' to equal 'July 17, 2015 3:02 AM'.
16:54:16 	    at /var/jenkins/workspace/pull-request-analyser-ng-simple@12/label/SLAVE/apps/files/tests/js/mainfileinfodetailviewSpec.js:54
```

Henni commented:
Awesome work!
Looking forward to adopt this in the tasks app.

blizzz commented:
I'll continue while @PVince81 is off next week.

So, If I understand you correct, you would introduce styles for a new #app-sidebar css class. It will take the style of mostly what is .detailsView right now, with adjustments to how it should behave – not overlapping, but reducing the size of the app-content respectively it's neighbours. Unfortunately I would not know how to modify css to get this behaved properly. I'd be happy about pointers from design gurus.

blizzz committed:
Main view does not commands detail and tab views to render. Fixes JS tests.

For a predictive behaviour we need to determine who is allowed to call render
methods on the views. Either, the main view is solely allowed to call render
and views do not do anything about output until then. Or, the main view relies
on the concrete views to be ready when things are about to be shown. The latter
approach has the advantage that concrete views know when they have to update
themselves (e.g. new data arrives or information change), but the main view
has now idea of the inner workings.

scrutinizer-notifier commented:
A [new inspection](https://scrutinizer-ci.com/g/owncloud/core/inspections/be853077-f7f2-4233-9bea-2caf8cc0dd62) was created.

jancborchardt commented:
@blizzz you can set the #app-content width to 70% while setting the #app-sidebar width to 30%. We do it similarly in the Mail app with the message list and the detail view. Let me know when you need help.

blizzz commented:
@jancborchardt code pointers would be nice. I tried to setup mail to see it in action, but cannot use it due to some Horde issues.

jancborchardt commented:
@blizzz setting up the Mail app should be no problem if you just clone it and pull in the dependencies via composer [as said in the readme](https://github.com/owncloud/mail/#developer-setup-info).

Then inspect it, that makes more sense than code pointers. In any case: https://github.com/owncloud/mail/blob/master/css/mail.css#L222-L223 (30% width for message list) and https://github.com/owncloud/mail/blob/master/css/mail.css#L551-L559 (70% width for detail view.

For this of course it should be main content 70% width (that is, 70% of the #app-content), and 30% width for the right sidebar.

blizzz committed:
extend MainFileInfoDetailView JS tests

scrutinizer-notifier commented:
A [new inspection](https://scrutinizer-ci.com/g/owncloud/core/inspections/830ce45f-ef24-4345-ac12-2d38bd69ce82) was created.

blizzz commented:
@jancborchardt i read readme, and that all went good, but then i am stopped at https://github.com/owncloud/mail/issues/897

Thanks for the pointers. I was having fun with JS Karma tests today :unamused: I will get the design change done tomorrow :yum:

blizzz commented:
Tests not working, because Jenkins, as far as i see.

blizzz committed:
sidebar shall not overlap but shrink  the contnet

scrutinizer-notifier commented:
A [new inspection](https://scrutinizer-ci.com/g/owncloud/core/inspections/22187603-4069-4820-abf2-273f4722fcbf) was created.

blizzz commented:
@jancborchardt ok, got  it so far. You may have a look? What's not good, yet, is it'll be too small on small screens, especially on mobile.  How was it supposed to behave in that case?

If the approach is ok, I'll transfer relevant parts to core apps css and js.

jancborchardt commented:
@blizzz how do I invoke it? Sharing still shows a dropdown, as does versions.

jancborchardt commented:
@blizzz regarding mobile screens, actually there we might need to have it overlap the content because of the limited width either way. Could be solvable by just CSS.

blizzz commented:
> how do I invoke it? Sharing still shows a dropdown, as does versions.

@jancborchardt click on an empty space on the file row. Where still on foundations. Sharing and Version will move and be replaced in a later PR.

> regarding mobile screens, actually there we might need to have it overlap the content because of the limited width either way. Could be solvable by just CSS.

Roger.

jancborchardt commented:
@blizzz some things:
* instead of having the class `detailsView` the sidebar should have the ID `app-sidebar` to fit with the rest. We do not use camelCase in classes or IDs.
* it should be over the controls bar as well, as in the prototype at https://github.com/owncloud/core/pull/16414#issue-77652540

blizzz commented:
> it should be over the controls bar as well, as in the prototype at #16414 (comment) 

I tend to agree, still having some doubts as the breadcrumbs might become inaccessible in a deep directory (and the other controls as well), requires the user to click the close-x when he is done what he wanted to do here. Assuming this will not happen too often indeed, it's probably reasonable.

blizzz committed:
rename sidebar element and use css and js utility functions to core

scrutinizer-notifier commented:
A [new inspection](https://scrutinizer-ci.com/g/owncloud/core/inspections/9f4126cc-ea2f-4da1-992c-fe85863f01a1) was created.

blizzz commented:
* .detailsView is now #app-sidebar and resides in core/css/apps.css
* showing and hiding functions reside in core/js/apps.js
* overlaps in mobile view
* I added also min-width in standard view. It will overlap if the sidebar would be smaller than 300px.

* [ ] FIXME:  when mobile.css is effective, click on file row downloads file, instead of opening sidebar

PVince81 commented:
If we want the new sidebar to be usable by any app, we need to move it into core.
Also we then need to think about how to register sections.
Because any app could register a section for specific app.
So if for example the news app had the sidebar, and another app wants to register an extra panel, it should then call something like `registerPanel("news", panelInstance)`.

blizzz commented:
We'll stick with having the basics in core and move styles and utility functions in a next iteration, after having collected experience and stabilized and polished what we're introducing here, so we can go with mature APIs later.

PVince81 commented:
Agreed with that.

blizzz committed:
fix show/hide app sidebar  functions

scrutinizer-notifier commented:
A [new inspection](https://scrutinizer-ci.com/g/owncloud/core/inspections/ed06953f-332e-4700-9747-e42dbcf85fad) was created.

MorrisJobke commented:
* [ ] please don't use the jQuery widget thingy. It's super ugly and doesn't really fit into the WebUI.

PVince81 commented:
@blizzz nice work on making the sidebar insert itself under the controls and shrink the list width.

However there seems to be a new issue:
Before scroll:
![rsidebar-top](https://cloud.githubusercontent.com/assets/277525/8958860/c687dabc-3607-11e5-8123-7a7629797376.png)

After scroll:
![rsidebar-top](https://cloud.githubusercontent.com/assets/277525/8958874/d8cd8564-3607-11e5-9ecf-277aabfc074a.png)

@jancborchardt should the sidebar stay sticky together with the controls bar then ?
This means that moving the scrollbar only ever scrolls the contents of the file list.

PVince81 commented:
- [ ] BUG: sidebar scrolls with contents

Commit:
Use own tabs impl in files sidebar

Also change style.

scrutinizer-notifier commented:
A [new inspection](https://scrutinizer-ci.com/g/owncloud/core/inspections/8838f4aa-72a8-4c49-af4a-46b66f4a4919) was created.

PVince81 commented:
Goodbye jquery.ui, now with own tabs implementation.


Please note that I added some generic styles in apps.css, so the tabs styles will be reusable: https://github.com/owncloud/core/commit/ef4c95dd02e53936f668fa9196d6401ba8a0a307#diff-d28ce211f72cb6409a66e6c8ab52d723R596

PVince81 commented:
![rsidebar-tabs](https://cloud.githubusercontent.com/assets/277525/8975589/23a9a6ac-3681-11e5-9b92-c8e40b790714.png)

jancborchardt committed:
fix right sidebar position and first style

jancborchardt committed:
fix position of elements in right sidebar

scrutinizer-notifier commented:
A [new inspection](https://scrutinizer-ci.com/g/owncloud/core/inspections/b3a95f61-7da7-44e9-9c81-87eb995ae7e2) was created.

jancborchardt commented:
Pushed some style fixes. The app-sidebar needs to be position: fixed. This leads to us not being able to properly size the app-content through CSS – we can only approach it, and do it via JS, similarly as we size the controls bar. (We can use the same function as for the controls bar sizing since we need to adjust that for the right sidebar anyway.)

jancborchardt commented:
Also, the favorite star toggle should work. :) Also, clicking on the thumbnail or the filename should open the viewer for the respective file. For folders, I’m not sure if clicking the thumbnail should do anything.

PVince81 commented:
@jancborchardt I'd like to do the favorite star separately because it needs some deep JS magic (interaction between sidebar and file model) and I don't want to do it in a hacky way. I might remove it / hide it before we merge this.

PVince81 commented:
@jancborchardt clicking on a folder icon would open the folder then ? The thumbnail would use the default action as registered by apps.

PVince81 commented:
- [ ] thumbnail click must trigger default action

oparoz commented:
> Also, clicking on the thumbnail or the filename should open the viewer for the respective file

Since the thumbnail offered is tiny, I would go with 2 different actions:
* One action to grow the thumbnail to 200x200, pushing the tabs down. This can be done with nice an overlay icon, like the one you see in video viewers when you want to make the viewport larger
* One action to open the file in it's default viewer. This can stay as the default action when clicking on the image

jancborchardt commented:
> I might remove it / hide it before we merge this.

@PVince81 yeah, probably best call about the fav star.

> clicking on a folder icon would open the folder then ? The thumbnail would use the default action as registered by apps.

Yup.

> Since the thumbnail offered is tiny, I would go with 2 different actions: One action to grow the thumbnail to 200x200, pushing the tabs down.

@oparoz not sure we want to go with a mode like that. I do agree though that the 50px thumbnail is a bit small for that we use it in a detail view. How about increasing the size to 75px?

PVince81 commented:
Hmmm, making the thumbnail clickable has the same issue as for the favorite star.
The problem is that the "FileActions" object is old and broken and isn't portable. It isn't enough to inject the file action object into all components, one would require to inject the FileList object and also the $tr element and pass it along to each view inside the sidebar.
Which means that in total we need to inject:
- FileInfo object (already there)
- FileActions object
- FileList object
- $tr object

This is too much, I'll try and find a way to "package" this or refactor it to a better class design.
Ideally we should only need to inject a single object into the view. Then the view will be able to read the FileInfo from that object and also trigger actions.

jancborchardt commented:
@PVince81 ok, making thumbnail interactive and duplicating the favorite icon there is not very important here because it’s duplication anyway – it’s mainly cherry on top.

What is important though for the sidebar is to put the sharing view into it, because without it it’s quite useless. ;) And then we need to continue with Versions, Activity & Comments.

blizzz commented:
Sharing will be a seperate PR.

PVince81 commented:
Before the sharing or other views can be added in we need to merge this branch first.

PVince81 commented:
@owncloud-bot retest this please

jancborchardt commented:
kk. @owncloud/designers please also give this a review! :)

PVince81 commented:
@jancborchardt the sidebar position and scrolling behavior is correct. However the main list's scrollbar isn't visible any more in Chromium. Did you intend the scrollbar to move on the blank space between the file list and scrollbar ? (might not be possible unless we change the scrollContainer for the list to not be the full page)

jancborchardt commented:
> Did you intend the scrollbar to move on the blank space between the file list and scrollbar ?

@PVince81 yeah, that’s probably what we need to do. Also, as said above the width of the container needs to be computed by JS as a result of the positioning fix – there should not be any blank space between the file list and the right sidebar of course.

If making the scrollbar display there is not possible, we can also of course absolutely position the app-sidebar a bit from the right (using getScrollbarWidth). But that will look strange once the sidebar gets longer and thus a scrollbar – resulting in confusing double scrollbar on the right.

PVince81 commented:
I'd still vote for merge first as it brings the base classes needed for further dev and fix the layout afterwards as it might take some time.

- [ ] fix layout / scrolling

Commit:
Fix app-content scrollbar when app-sidebar is open

scrutinizer-notifier commented:
A [new inspection](https://scrutinizer-ci.com/g/owncloud/core/inspections/804edb14-91c3-49f9-aa8a-356fc7aba9cc) was created.

PVince81 commented:
@jancborchardt I found a non-intrusive fix that doesn't require revisiting the DOM layout.
See https://github.com/owncloud/core/commit/abe2213ba98c84ec03b877d2fa6a852f5203a291

Basically the sidebar width is set to 27% and the margin-right of app-content set to 27% too...

PVince81 commented:
Jenkins PR: https://github.com/owncloud/core/pull/18103

PVince81 commented:
Looks like these tests are too time-dependent:
```
Expected '幾秒前' to equal 'a few seconds ago'.
09:38:00 	    at /var/jenkins/workspace/pull-request-analyser-ng-simple@2/label/SLAVE/apps/files/tests/js/mainfileinfodetailviewSpec.js:56
09:38:00 	Expected '2015年7月16日晚上9點02分' to equal 'July 17, 2015 3:02 AM'.
09:38:00 	    at /var/jenkins/workspace/pull-request-analyser-ng-simple@2/label/SLAVE/apps/files/tests/js/mainfileinfodetailviewSpec.js:57
```

I'll have a look.

PVince81 commented:
Weird, looks like something put the test server in a different timezone and locale !? (taiwanese)

Commit:
Set momentJS locale to en during unit test runs

scrutinizer-notifier commented:
A [new inspection](https://scrutinizer-ci.com/g/owncloud/core/inspections/090be754-5450-4451-8636-83ee615337ad) was created.

PVince81 commented:
@jancborchardt I also wonder whether the sidebar should also open when clicking on the other columns like mtime and size ?

PVince81 commented:
Ok now fighting against a timezone issue:
```
10:59:42 PhantomJS 1.9.8 (Linux 0.0.0) OCA.Files.MainFileInfoDetailView tests rendering displays basic info FAILED
10:59:42 	Expected 'July 16, 2015 9:02 PM' to equal 'July 17, 2015 3:02 AM'.
10:59:42 	    at /var/jenkins/workspace/pull-request-analyser-ng-simple@2/label/SLAVE/apps/files/tests/js/mainfileinfodetailviewSpec.js:57
```

Commit:
Fix sidebar mtime display unit test

scrutinizer-notifier commented:
A [new inspection](https://scrutinizer-ci.com/g/owncloud/core/inspections/f3bb2c06-0844-4122-89d4-8ea5bccc530d) was created.

PVince81 commented:
Hmmm, same problem...
```
12:35:39 PhantomJS 1.9.8 (Linux 0.0.0) OCA.Files.MainFileInfoDetailView tests rendering displays basic info FAILED
12:35:39 	Expected 'July 16, 2015 9:02 PM' to equal 'July 17, 2015 3:02 AM'.
12:35:39 	    at /var/jenkins/workspace/pull-request-analyser-ng-simple@4/label/SLAVE/apps/files/tests/js/mainfileinfodetailviewSpec.js:57
```

blizzz committed:
fix timezone issue with unittest

blizzz committed:
fix tipsy gravity after rebase

scrutinizer-notifier commented:
A [new inspection](https://scrutinizer-ci.com/g/owncloud/core/inspections/114fbf35-e122-4d27-b4ef-a65833833bf0) was created.

blizzz commented:
My attempt to fix the test. Also rebased.

Henni committed:
replace tipsy with tooltip

scrutinizer-notifier commented:
A [new inspection](https://scrutinizer-ci.com/g/owncloud/core/inspections/7717e9c7-b992-4f64-86be-158381e74de1) was created.

PVince81 commented:
JS unit tests fail:
```
07:24:11 	Expected '' to equal 'One.txt'.
07:24:11 	    at /var/jenkins/workspace/pull-request-analyser-ng-simple/label/SLAVE/apps/files/tests/js/mainfileinfodetailviewSpec.js:54
07:24:11 	Expected '' to equal '123456789 bytes'.
07:24:11 	    at /var/jenkins/workspace/pull-request-analyser-ng-simple/label/SLAVE/apps/files/tests/js/mainfileinfodetailviewSpec.js:56
07:24:11 	Expected '' to equal 'July 16, 2015 9:02 PM'.
07:24:11 	    at /var/jenkins/workspace/pull-request-analyser-ng-simple/label/SLAVE/apps/files/tests/js/mainfileinfodetailviewSpec.js:58
```

I'll look into this.

Commit:
Fix unit test for tipsy->tooltip change

PVince81 commented:
I hereby approve all commits from @blizzz and @Henni and @jancborchardt with this :+1: 

Please review my commits too so we can get this merged.

blizzz commented:
And so am I honoured to give a :+1: to all commits from @PVince81 @Henni and @jancborchardt :fireworks:

PVince81 commented:
Unless there is any objection, I will merge this as soon as Jenkins gives the green light. :bulb:

PVince81 commented:
Here we go.

Please raise tickets separately.

I'll soon make a ticket for the file actions (favorites + click on thumbnail)

PVince81 commented:
Raised ticket to implement the missing actions (thumbnail click + favorite click): https://github.com/owncloud/core/issues/18169

jancborchardt commented:
@PVince81 great work man! :)

DeepDiver1975 commented:
@PVince81 branch can be deleted?

PVince81 commented:
>  branch can be deleted?

Yes. Strangely Github didn't show me the button yesterday.

MorrisJobke commented:
> Yes. Strangely Github didn't show me the button yesterday.

On mobile? Because on mobile it isn't present

PVince81 commented:
No, on desktop web UI. It was probably a temporary glitch.
